x-pack/test/alerting_api_integration/observability/helpers/syntrace.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import {
+  ApmSynthtraceEsClient,
+  ApmSynthtraceKibanaClient,
+  createLogger,
+  LogLevel,
+} from '@kbn/apm-synthtrace';
+import { apm, timerange } from '@kbn/apm-synthtrace-client';
+
+export const getSyntraceClient = async ({
+  kibanaUrl,
+  esClient,
+}: {
+  kibanaUrl: string;
+  esClient: Client;
+}) => {
+  const kibanaClient = new ApmSynthtraceKibanaClient({
```

Comment:
@fkanout maybe I am missing something, but is there a reason we are using APM data generator for Threshold rule (intended for infra data). How about reusing https://github.com/elastic/kibana/pull/157489/files#diff-3ae1e445848705de49b07c6c68a528947fa96cb61f52e6ea02ee86ac8c691ae5?